File: src/Flagrum.Core/Graphics/Models/GameModelMeshPart.cs
using Flagrum.Core.Serialization.MessagePack;

namespace Flagrum.Core.Graphics.Models;

public class GameModelMeshPart : IMessagePackItem
{
    public uint PartsId { get; set; }
    public uint StartIndex { get; set; }
    public uint IndexCount { get; set; }

    public void Read(MessagePackReader reader)
    {
        PartsId = reader.Read<uint>();
        StartIndex = reader.Read<uint>();
        IndexCount = reader.Read<uint>();
    }

    public void Write(MessagePackWriter writer)
    {
        writer.Write(PartsId);
        writer.Write(StartIndex);
        writer.Write(IndexCount);
    }
}

File: src/Flagrum.Blender/core/graphics/models/gamemodelmeshpart.py
from dataclasses import dataclass

from ...serialization.messagepack.reader import MessagePackReader
from ...serialization.messagepack.writer import MessagePackWriter


@dataclass(init=False)
class GameModelMeshPart:
    parts_id: int
    start_index: int
    index_count: int

    def read(self, reader: MessagePackReader):
        self.parts_id = reader.read()
        self.start_index = reader.read()
        self.index_count = reader.read()

    def writer(self, writer: MessagePackWriter):
        writer.write(self.parts_id)
        writer.write(self.start_index)
        writer.write(self.index_count)
